operations/ftdc.go

```diff
@@ -737,19 +737,82 @@ func toT2() cli.Command {
 			ctx, cancel := context.WithCancel(context.Background())
 			defer cancel()
 
-			inputFile, err := os.Open(inputPath)
+			inputStat, err := os.Stat(inputPath)
 			if err != nil {
 				return errors.Wrapf(err, "problem opening file '%s'", inputPath)
 			}
-			defer func() { grip.Warning(inputFile.Close()) }()
 
-			// All genny output files are named using the workload actor and operation.
-			//   i.e., Actor.Operation.ftdc
-			//
-			// We get the actor and operation names from the ftdc filepath for use
-			// in the translation process.
-			fileName := filepath.Base(inputPath)
-			actorOperation := strings.Split(fileName, ".ftdc")
+			// Stores ftdc chunk iterators and its metadata to be used in TranslateGenny.
+			var outputSlice []*ftdc.GennyOutputMetadata
+
+			// Check if input path is an individual file or directory.
+			// All ftdc files in the provided input directory are initially
+			// used to set GennyOutputMetadata Name, Iter, StartTime and EndTime.
+			switch mode := inputStat.Mode(); {
+			// Handle directory.
+			case mode.IsDir():
+				if !strings.HasSuffix(inputPath, "/") {
+					inputPath += "/"
+				}
+				input, err := os.Open(inputPath)
+				if err != nil {
+					return errors.Wrapf(err, "problem opening input path '%s'", inputPath)
+				}
+				files, err := input.Readdir(-1)
+				if err != nil {
+					return errors.Wrapf(err, "problem reading dir '%s'", inputPath)
+				}
+
+				for _, file := range files {
+					if file.Mode().IsRegular() && filepath.Ext(file.Name()) == ".ftdc" {
+						var gennyOutput ftdc.GennyOutputMetadata
+
+						f, err := os.Open(inputPath + file.Name())
+						if err != nil {
+							return errors.Wrapf(err, "problem opening file '%s'", inputPath+file.Name())
+						}
+						defer func() { grip.Warning(f.Close()) }()
+						gennyOutput.Iter = ftdc.ReadChunks(ctx, f)
+						gennyOutput = ftdc.GetGennyTime(ctx, gennyOutput)
+
+						// Reset the chunk iterator for the ftdc file.
+						input.Close()
```

Comment:
Not sure I understand what is going on here. Are you purposely opening the same file and reading it twice? Also why do you close `input` here?

Reply:
I'm purposefully opening the same file and reading it twice. The first time is to gather the start and end of the individual operation and update the genny metadata after doing so. At this point, the iterator is exhausted so I reopen the file to start the iterator back at the beginning.

Reply:
I don't have a reason for repeatedly closing the file though.